# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/parent/FileSystemAccessHandleControlParent.h
# Commit: b68e19021833
# Full Hash: b68e1902183357e129a1a53490447fc3cc219f4f
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2023-02-07 16:42:31
# Regressor Bug: 1809064
# File Overlap Count: 1
# Description:
#   Bug 1809064 - Add PFileSystemAccessHandleControl top level protocol; r=dom-storage-reviewers,jari
#   
#   Sync access handle methods need to run sync loops to block the thread until a
#   result is available. The main protocol PFileSystemManager and its subprotocol
#   PFileSystemAccessHandle is bound to the main worker event queue, so incoming
# ==============================================================================

diff -r fc3ab5ee8ef3 -r b68e19021833 dom/fs/parent/FileSystemAccessHandleControlParent.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/fs/parent/FileSystemAccessHandleControlParent.h	Tue Feb 07 08:27:06 2023 +0000
@@ -0,0 +1,43 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef DOM_FS_PARENT_FILESYSTEMACCESSHANDLECONTROLPARENT_H_
+#define DOM_FS_PARENT_FILESYSTEMACCESSHANDLECONTROLPARENT_H_
+
+#include "mozilla/dom/PFileSystemAccessHandleControlParent.h"
+#include "nsISupportsUtils.h"
+
+namespace mozilla::dom {
+
+class FileSystemAccessHandle;
+
+class FileSystemAccessHandleControlParent
+    : public PFileSystemAccessHandleControlParent {
+ public:
+  explicit FileSystemAccessHandleControlParent(
+      RefPtr<FileSystemAccessHandle> aAccessHandle);
+
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(FileSystemAccessHandleControlParent,
+                                        override)
+
+  mozilla::ipc::IPCResult RecvFoo();
+
+  void ActorDestroy(ActorDestroyReason aWhy) override;
+
+ protected:
+  virtual ~FileSystemAccessHandleControlParent();
+
+ private:
+  RefPtr<FileSystemAccessHandle> mAccessHandle;
+
+#ifdef DEBUG
+  bool mActorDestroyed = false;
+#endif
+};
+
+}  // namespace mozilla::dom
+
+#endif  // DOM_FS_PARENT_FILESYSTEMACCESSHANDLECONTROLPARENT_H_